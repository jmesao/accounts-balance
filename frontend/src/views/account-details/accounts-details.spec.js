import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
// TODO.- Mock vue-socket.io module and add more tests
import VueSocketIO from 'vue-socket.io';

import AccountDetailsView from './account-details.vue';

describe('AccountDetailsView', () => {
  function mountAccountDetailsView(mountOpts) {
    const localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(new VueSocketIO({
      debug: true,
      connection: 'http://localhost',
    }));

    const store = new Vuex.Store({
      state: {},
      mutations: {},
    });

    return shallowMount(AccountDetailsView, {
      localVue,
      ...mountOpts,
      store,
    });
  }

  it('should render without account nor items on the account statements table', () => {
    const wrapper = mountAccountDetailsView({
      mocks: {
        $route: {
          name: 'account-details',
          params: {
            id: 1,
            account: null,
          },
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
