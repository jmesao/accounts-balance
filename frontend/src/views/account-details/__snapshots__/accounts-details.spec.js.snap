// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetailsView should render without account nor items on the account statements table 1`] = `
<div>
  <header-container-stub
    title="Account detail"
  >
    <div>
      Home / Accounts / 
      <span
        class="tw-body-strong"
      >
        Details
      </span>
    </div>
  </header-container-stub>
   
  <div
    class="card-container ec-card"
  >
    <div>
      <div
        class="tw-h4"
      >
        Dummy Test Name
      </div>
       
      <div
        class="tw-small-text"
      >
        Treasury account
      </div>
    </div>
  </div>
   
  <account-statements-table-stub
    items=""
  />
</div>
`;
